avalon/tools/cbsceneinventory/proxy.py

```diff
@@ -6,32 +6,48 @@ class RecursiveSortFilterProxyModel(QtCore.QSortFilterProxyModel):
     """Filters to the regex if any of the children matches allow parent"""
 
     def filterAcceptsRow(self, row, parent):
+        if not self.filterRegExp().isEmpty():
 
-        regex = self.filterRegExp()
-        if not regex.isEmpty():
-            pattern = regex.pattern()
-            pattern = re.escape(pattern)
+            pattern = re.escape(self.filterRegExp().pattern())
             model = self.sourceModel()
             source_index = model.index(row, self.filterKeyColumn(), parent)
+
             if source_index.isValid():
 
                 # Check current index itself
                 key = model.data(source_index, self.filterRole())
                 if re.search(pattern, key, re.IGNORECASE):
                     return True
 
-                # Check children
                 rows = model.rowCount(source_index)
-                for i in range(rows):
-                    if self.filterAcceptsRow(i, source_index):
-                        return True
+                if not rows:
+                    # Display item if parent matches
+                    return True
 
-                # Otherwise filter it
-                return False
+                return any([self.filterAcceptChildRow(i, source_index, pattern)
+                            for i in range(rows)])
 
         return super(RecursiveSortFilterProxyModel,
                      self).filterAcceptsRow(row, parent)
 
+    def filterAcceptChildRow(self, row, parent, pattern):
```

Comment:
Naming of the method might confuse users that it's a default Qt method, it's not. Maybe avoid the upper camel case here?